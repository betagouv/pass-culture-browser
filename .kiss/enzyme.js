import React from 'react'
import { shallow } from 'enzyme'

import MyComponent from '../MyComponent'

describe('src | components | MyComponent', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      const props = { errors: {} }
      const wrapper = shallow(<MyComponent {...props} />)
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
