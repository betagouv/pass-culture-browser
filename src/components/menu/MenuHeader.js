/* eslint
  react/jsx-one-expression-per-line: 0 */
import PropTypes from 'prop-types'
import React from 'react'
import { connect } from 'react-redux'

import { selectCurrentUser } from '../hocs'
import { ROOT_PATH } from '../../utils/config'
import { getWalletValue } from '../../utils/user'

export const RawMenuHeader = ({ currentUser }) => {
  const walletValue = getWalletValue(currentUser)
  const avatar = `${ROOT_PATH}/icons/avatar-default-w-XL.svg`
  return (
    <div id="main-menu-header" className="flex-columns is-relative py16 fs18">
      <div className="column-profile text-center px18">
        <p id="main-menu-header-avatar">
          <img alt="" src={avatar} className="mb3" />
        </p>
        <p id="main-menu-header-username" className="px5 is-hyphens">
          <span className="fs16 is-medium">
            {currentUser && currentUser.publicName}
          </span>
        </p>
      </div>
      <div className="column-account flex-1 flex-rows flex-center px18">
        <p className="fs30">
          <span>Mon Pass</span>
        </p>
        <p id="main-menu-header-wallet-value">
          <span
            className="fs44 is-semi-bold"
            style={{ letterSpacing: '-0.1rem', lineHeight: '42px' }}
          >
            {walletValue}
          </span>
          <span className="fs30" style={{ lineHeight: '42px' }}>
            &nbsp;â‚¬
          </span>
        </p>
      </div>
    </div>
  )
}

RawMenuHeader.defaultProps = {
  currentUser: null,
}

RawMenuHeader.propTypes = {
  currentUser: PropTypes.oneOfType([PropTypes.bool, PropTypes.object]),
}

function mapStateToProps(state) {
  return {
    currentUser: selectCurrentUser(state),
  }
}

export default connect(mapStateToProps)(RawMenuHeader)
