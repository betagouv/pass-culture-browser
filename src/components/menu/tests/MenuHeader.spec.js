// jest --env=jsdom ./src/components/menu/tests/MenuHeader --watch
import React from 'react'
import { shallow } from 'enzyme'

import { RawMenuHeader } from '../MenuHeader'

const walletValueId = '#main-menu-header-wallet-value'

describe('src | components | menu | MenuHeader', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      // given
      const props = {
        currentUser: {
          publicName: 'username',
          wallet_balance: 500,
        },
      }

      // when
      const wrapper = shallow(<RawMenuHeader {...props} />)

      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
  describe('render', () => {
    it('with no user', () => {
      // given
      const props = {}
      // when
      const wrapper = shallow(<RawMenuHeader {...props} />)
      const walletElement = wrapper.find(walletValueId).first()
      // then
      const expected = '--\u00A0€'
      const text = walletElement.text()
      expect(text).toEqual(expected)
    })
    it('with user but no wallet_balance', () => {
      // given
      const props = { currentUser: {} }
      // when
      const wrapper = shallow(<RawMenuHeader {...props} />)
      const walletElement = wrapper.find(walletValueId).first()
      // then
      const expected = '--\u00A0€'
      const text = walletElement.text()
      expect(text).toEqual(expected)
    })
    it('with user and wallet_balance', () => {
      // given
      const props = { currentUser: { wallet_balance: 500 } }
      // when
      const wrapper = shallow(<RawMenuHeader {...props} />)
      const walletElement = wrapper.find(walletValueId).first()
      // then
      const expected = '500\u00A0€'
      const text = walletElement.text()
      expect(text).toEqual(expected)
    })
  })
})
