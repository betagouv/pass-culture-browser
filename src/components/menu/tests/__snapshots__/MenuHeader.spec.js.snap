// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | menu | MenuHeader snapshot should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RawMenuHeader
    currentUser={
      Object {
        "publicName": "username",
        "wallet_balance": 500,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="column-profile text-center"
        >
          <p
            id="main-menu-header-avatar"
          >
            <img
              alt=""
              className="mb3"
              src="http://localhost/icons/avatar-default-w-XL.svg"
            />
          </p>
          <p
            className="is-clipped text-ellipsis px5"
            id="main-menu-header-username"
          >
            <span>
              username
            </span>
          </p>
        </div>,
        <div
          className="column-account flex-1 flex-rows flex-center px12"
        >
          <p
            className="fs30"
          >
            <span>
              Mon Pass
            </span>
          </p>
          <p
            className="text-center"
            id="main-menu-header-wallet-value"
          >
            <span
              className="fs39 is-semi-bold"
              style={
                Object {
                  "lineHeight": "42px",
                }
              }
            >
              500
               €
            </span>
          </p>
        </div>,
      ],
      "className": "flex-columns is-relative py16 fs18",
      "id": "main-menu-header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              id="main-menu-header-avatar"
            >
              <img
                alt=""
                className="mb3"
                src="http://localhost/icons/avatar-default-w-XL.svg"
              />
            </p>,
            <p
              className="is-clipped text-ellipsis px5"
              id="main-menu-header-username"
            >
              <span>
                username
              </span>
            </p>,
          ],
          "className": "column-profile text-center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt=""
                className="mb3"
                src="http://localhost/icons/avatar-default-w-XL.svg"
              />,
              "id": "main-menu-header-avatar",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "className": "mb3",
                "src": "http://localhost/icons/avatar-default-w-XL.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span>
                username
              </span>,
              "className": "is-clipped text-ellipsis px5",
              "id": "main-menu-header-username",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "username",
              },
              "ref": null,
              "rendered": "username",
              "type": "span",
            },
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="fs30"
            >
              <span>
                Mon Pass
              </span>
            </p>,
            <p
              className="text-center"
              id="main-menu-header-wallet-value"
            >
              <span
                className="fs39 is-semi-bold"
                style={
                  Object {
                    "lineHeight": "42px",
                  }
                }
              >
                500
                 €
              </span>
            </p>,
          ],
          "className": "column-account flex-1 flex-rows flex-center px12",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span>
                Mon Pass
              </span>,
              "className": "fs30",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Mon Pass",
              },
              "ref": null,
              "rendered": "Mon Pass",
              "type": "span",
            },
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span
                className="fs39 is-semi-bold"
                style={
                  Object {
                    "lineHeight": "42px",
                  }
                }
              >
                500
                 €
              </span>,
              "className": "text-center",
              "id": "main-menu-header-wallet-value",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  500,
                  " €",
                ],
                "className": "fs39 is-semi-bold",
                "style": Object {
                  "lineHeight": "42px",
                },
              },
              "ref": null,
              "rendered": Array [
                500,
                " €",
              ],
              "type": "span",
            },
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="column-profile text-center"
          >
            <p
              id="main-menu-header-avatar"
            >
              <img
                alt=""
                className="mb3"
                src="http://localhost/icons/avatar-default-w-XL.svg"
              />
            </p>
            <p
              className="is-clipped text-ellipsis px5"
              id="main-menu-header-username"
            >
              <span>
                username
              </span>
            </p>
          </div>,
          <div
            className="column-account flex-1 flex-rows flex-center px12"
          >
            <p
              className="fs30"
            >
              <span>
                Mon Pass
              </span>
            </p>
            <p
              className="text-center"
              id="main-menu-header-wallet-value"
            >
              <span
                className="fs39 is-semi-bold"
                style={
                  Object {
                    "lineHeight": "42px",
                  }
                }
              >
                500
                 €
              </span>
            </p>
          </div>,
        ],
        "className": "flex-columns is-relative py16 fs18",
        "id": "main-menu-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                id="main-menu-header-avatar"
              >
                <img
                  alt=""
                  className="mb3"
                  src="http://localhost/icons/avatar-default-w-XL.svg"
                />
              </p>,
              <p
                className="is-clipped text-ellipsis px5"
                id="main-menu-header-username"
              >
                <span>
                  username
                </span>
              </p>,
            ],
            "className": "column-profile text-center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt=""
                  className="mb3"
                  src="http://localhost/icons/avatar-default-w-XL.svg"
                />,
                "id": "main-menu-header-avatar",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "className": "mb3",
                  "src": "http://localhost/icons/avatar-default-w-XL.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  username
                </span>,
                "className": "is-clipped text-ellipsis px5",
                "id": "main-menu-header-username",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "username",
                },
                "ref": null,
                "rendered": "username",
                "type": "span",
              },
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="fs30"
              >
                <span>
                  Mon Pass
                </span>
              </p>,
              <p
                className="text-center"
                id="main-menu-header-wallet-value"
              >
                <span
                  className="fs39 is-semi-bold"
                  style={
                    Object {
                      "lineHeight": "42px",
                    }
                  }
                >
                  500
                   €
                </span>
              </p>,
            ],
            "className": "column-account flex-1 flex-rows flex-center px12",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  Mon Pass
                </span>,
                "className": "fs30",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Mon Pass",
                },
                "ref": null,
                "rendered": "Mon Pass",
                "type": "span",
              },
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span
                  className="fs39 is-semi-bold"
                  style={
                    Object {
                      "lineHeight": "42px",
                    }
                  }
                >
                  500
                   €
                </span>,
                "className": "text-center",
                "id": "main-menu-header-wallet-value",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    500,
                    " €",
                  ],
                  "className": "fs39 is-semi-bold",
                  "style": Object {
                    "lineHeight": "42px",
                  },
                },
                "ref": null,
                "rendered": Array [
                  500,
                  " €",
                ],
                "type": "span",
              },
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
